test/helpers/kubectl.go

```diff
@@ -3654,12 +3654,13 @@ func (kub *Kubectl) GetCiliumPodOnNode(node string) (string, error) {
 
 // GetNodeInfo provides the node name and IP address based on the label
 // (eg helpers.K8s1 or helpers.K8s2)
-func (kub *Kubectl) GetNodeInfo(label string) (nodeName, nodeIP string) {
+func (kub *Kubectl) GetNodeInfo(label string) (nodeName, nodeInternalIP, nodeExternalIP string) {
 	nodeName, err := kub.GetNodeNameByLabel(label)
 	gomega.ExpectWithOffset(1, err).To(gomega.BeNil(), "Cannot get node by label "+label)
-	nodeIP, err = kub.GetNodeIPByLabel(label, false)
-	gomega.ExpectWithOffset(1, err).Should(gomega.BeNil(), "Can not retrieve Node IP for "+label)
-	return nodeName, nodeIP
+	nodeInternalIP, err = kub.GetNodeIPByLabel(label, false)
+	gomega.ExpectWithOffset(1, err).Should(gomega.BeNil(), "Can not retrieve Node Internal IP for "+label)
+	nodeExternalIP, _ = kub.GetNodeIPByLabel(label, true)
```

Comment:
If this ever fails it will be pretty hard to figure it out. Please handle the error with `gomega.ExpectWithOffset(1, err).Should(gomega.BeNil(), "Can not retrieve Node External IP for "+label)`, same as is done for internal IP.

Reply:
I'm not sure if we should fail in case we can't retrieve ExternalIP. Does all the integration tests include kubernetes node with an external IP?

Reply:
We return an error if a node does not have any external IP assigned. If all k8s nodes in our integration test has an ExternalIP, then it is safe to fail here. Otherwise this should be fine.

https://github.com/cilium/cilium/blob/e567f8bfa7e8cc4d71566f224f60d7222d3d874b/test/helpers/kubectl.go#L1094-L1097

Reply:
These tests can be run on any k8s cluster that is set up to run them ( https://docs.cilium.io/en/v1.9/contributing/testing/e2e/#running-end-to-end-tests-in-other-environments-via-kubeconfig ), lack of handling this error potentially costs a developer debugging the test on a cluster without external IPs non-trivial amount of time. If we just fail the test here saying "there are no external IPs and there should be", we give the developer a clear message on what is wrong with their setup.

Reply:
The issue is `GetNodeInfo` function is used in many tests including ones that don't need k8s node with externalIPs. So my question was whether all nodes in CIlium CI jobs have external IPs? If not, then the tests will fail even if they didn't need k8s node with externalIPs.

Reply:
Ah, I see what you mean. In this case, we should log a warning that the external IP was not retrieved, silently ignoring errors will lead to a lot of frustrations.

Reply:
Thanks. I changed things a bit. I try to retrieve ExternalIP of the Node only when we run on GKE as that's when we need it. I fail the test if we can't retrieve it.